### PR TITLE
Allow to provide an `--inputs-from` argument when using flakes 

### PR DESCRIPTION
**Pull request by madjar**
This allows pinning the nixpkgs to use when there exist a flake for the current system
(for example when using home-manager).

I realized it would be useful to me to reuse my home-manager config's nixpkgs instead of downloading a new one anytime the cache expires.

My rust is not in top shape, so I'm definitely open to comments on the style of the code (or in fixes, if you have some in mind).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@ result
 # Added by cargo
 
 /target
+
+.direnv/
```
Comment (SuperSandro2000) [inline .gitignore:9]: Missing final new line. Also we could safely remove all the other new lines.

**Review by SuperSandro2000 — COMMENTED** [review]
I don't think that extra indirection is really what we want. Why not let people overwrite the nixpkgs channel with a flake-url?

File: .gitignore (hunk 1 above)
Comment (madjar) [reply .gitignore]: Thanks, the final newline has been added.

**Comment by madjar** [comment]
That sounds like a good idea, but I'm not aware of ways to use flakes as channels. Do you have something specific in mind?

**Comment by Artturin** [comment]
> This allows pinning the nixpkgs to use when there exist a flake for the current system
(for example when using home-manager).

> I realized it would be useful to me to reuse my home-manager config's nixpkgs instead of downloading a new one anytime the cache expires.

this is already possible with 

```nix
  nix = {
    registry = {
      nixpkgs.flake = { outPath = lib.cleanSource "${pkgs.path}"; };
    };
  };
```
in nixos and home-manager https://nix-community.github.io/home-manager/options.html#opt-nix.registry

**Pull request closed without merging**